lib/sync.js

```diff
@@ -133,11 +142,7 @@ module.exports = function resolveSync(x, options) {
             return loadpkg(path.dirname(dir));
```

Comment:
@SimenB Separately from this PR, it would also be nice if there was a way to short-circuit this recursive check for package.json file in every directory.

Reply:
agreed, like the `findClosestPackageFile` I mentioned in https://github.com/facebook/jest/issues/11034#issuecomment-770194055?

Reply:
Yeah!

Reply:
I'm open to a PR that enables that, for sure.